testlib/resources.go

```diff
@@ -0,0 +1,169 @@
+			if found == false {
+				panic(fmt.Sprintf("Failed to find folder %s", testResource.src))
+			}
+
+			testResourcesToSetup[i].src = srcPath
+		} else {
+			panic(fmt.Sprintf("Invalid resource type: %d", testResource.resType))
+		}
+	}
+}
+
+func SetupTestResources() (string, error) {
+	resolveResourcesPath()
+
+	tempDir, err := ioutil.TempDir("", "testlib")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create temporary directory")
+	}
+
+	pluginsDir := path.Join(tempDir, "plugins")
+	err = os.Mkdir(pluginsDir, 0700)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create plugins directory %s", pluginsDir)
+	}
+
+	// Symlinking cwd(package path) to temp for tests which need to run from that path.
+	// compileGo is mostly only function which needs this to be able to find packages while compiling code
+	pkgDir := path.Join(tempDir, "pkg")
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to get current working directory")
+	}
+
+	err = os.Symlink(cwd, pkgDir)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to symlink %s to %s", cwd, pkgDir)
+	}
+
+	err = setupConfig(path.Join(tempDir, "config"))
+	if err != nil {
+		return "", errors.Wrap(err, "failed to setup config")
+	}
+
+	var resourceDestInTemp string
+
+	// Setting up test resources in temp.
+	// Action in each resource tells whether it needs to be copied or just symlinked
+	for _, testResource := range testResourcesToSetup {
+		resourceDestInTemp = filepath.Join(tempDir, testResource.dest)
+
+		if testResource.action == actionCopy {
+			if testResource.resType == resourceTypeFile {
+				err = utils.CopyFile(testResource.src, resourceDestInTemp)
+				if err != nil {
+					return "", errors.Wrapf(err, "failed to copy file %s to %s", testResource.src, resourceDestInTemp)
+				}
+			} else if testResource.resType == resourceTypeFolder {
+				err = utils.CopyDir(testResource.src, resourceDestInTemp)
+				if err != nil {
+					return "", errors.Wrapf(err, "failed to copy folder %s to %s", testResource.src, resourceDestInTemp)
+				}
+			}
+		} else if testResource.action == actionSymlink {
+			destDir := path.Dir(resourceDestInTemp)
+			if destDir != "." {
+				err = os.MkdirAll(destDir, os.ModePerm)
+				if err != nil {
+					return "", errors.Wrapf(err, "failed to make dir %s", destDir)
+				}
+			}
+
+			err = os.Symlink(testResource.src, resourceDestInTemp)
+			if err != nil {
+				return "", errors.Wrapf(err, "failed to symlink %s to %s", testResource.src, resourceDestInTemp)
+			}
+		} else {
+			return "", errors.Wrapf(err, "Invalid action: %d", testResource.action)
+		}
+
+	}
+
+	return tempDir, nil
+}
+
+func setupConfig(configDir string) error {
+	var err error
+	var config model.Config
+
+	config.SetDefaults()
+
+	err = os.Mkdir(configDir, 0700)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create config directory %s", configDir)
+	}
+
+	defaultJson := path.Join(configDir, "default.json")
```

Comment:
Separately, I'm making some other improvements that should render the need to copy/generate `default.json` unnecessary altogether. I'll circle back here once those are done.